auth/src/main/java/com/firebase/ui/auth/util/SmartLock.java

```diff
@@ -201,7 +205,11 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
     }
 
     private void finish(int resultCode) {
-        mActivity.finish(RESULT_OK, mActivity.getIntent());
+        mActivity.finish(resultCode, mActivity.getIntent());
+    }
+
+    private void finish(int resultCode, Intent data) {
+        mActivity.finish(resultCode, data);
```

Comment:
Since this is probably going to be a point of confusion in the future, I think we should just remove the option to set a result at all and always pass down RESULT_OK. What do you think?